src/android/PendingUploadDao.java

```diff
@@ -0,0 +1,53 @@
+package com.spoon.backgroundfileupload;
+
+import androidx.room.Dao;
+import androidx.room.Delete;
+import androidx.room.Insert;
+import androidx.room.OnConflictStrategy;
+import androidx.room.Query;
+
+import java.util.List;
+
+@Dao
+public interface PendingUploadDao {
+    @Query("SELECT * FROM pending_upload")
+    List<PendingUpload> getAll();
+
+    @Query("SELECT COUNT(*) FROM pending_upload")
+    int getAllCount();
+
+    @Query("SELECT * FROM pending_upload WHERE id = :id")
+    PendingUpload getById(final String id);
+
+    @Query("SELECT * FROM pending_upload WHERE state = 'PENDING' LIMIT 1")
+    PendingUpload getFirstPendingEntry();
+
+    @Query("SELECT COUNT(*) FROM pending_upload WHERE state = 'PENDING'")
+    int getPendingUploadsCount();
+
+    @Query("SELECT * FROM pending_upload WHERE state = 'UPLOADED'")
+    int getCompletedUploads();
+
+    @Query("SELECT COUNT(*) FROM pending_upload WHERE state = 'UPLOADED'")
+    int getCompletedUploadsCount();
+
+    @Query("UPDATE pending_upload SET state = 'UPLOADED' WHERE ID = :id")
+    void markAsUploaded(final String id);
+
+    default boolean exists(final String id) {
+        return getById(id) != null;
```

Comment:
is a `@Query(select count(id) where id = id) > 0 ?` better than fetching the whole row in this case, as we are just checking if it exists or not ?